keymap_drawer/draw.py

```diff
@@ -22,34 +23,91 @@ def __init__(self, config: DrawConfig, out: TextIO, **kwargs) -> None:
 
     @staticmethod
     def _split_text(text: str) -> list[str]:
         # do not split on double spaces, but do split on single
         return [word.replace("\x00", " ") for word in text.replace("  ", "\x00").split()]
 
-    def _draw_rect(self, p: Point, w: float, h: float, cls: Sequence[str]) -> None:
+    def _draw_rect(self, p: Point, w: float, h: float, cls: Sequence[str],) -> None:
         class_str = (' class="' + " ".join(c for c in cls if c) + '"') if cls else ""
         self.out.write(
             f'<rect rx="{self.cfg.key_rx}" ry="{self.cfg.key_ry}" x="{p.x - w / 2}" y="{p.y - h / 2}" '
             f'width="{w}" height="{h}"{class_str}/>\n'
         )
 
-    def _draw_text(self, p: Point, words: Sequence[str], cls: Sequence[str], shift: float = 0) -> None:
+    def _draw_text(self, p: Point, words: Sequence[str], cls: Sequence[str], shift: float = 0, legend_type: str="") -> None:
         if not words or not words[0]:
             return
 
+        if glyph:= self._is_glyph(words):
+            if self._draw_glyph(p, glyph, cls, legend_type):
+                return
+
         class_str = (' class="' + " ".join(c for c in cls if c) + '"') if cls else ""
+
         if len(words) == 1:
             self.out.write(f'<text x="{p.x}" y="{p.y}"{class_str}>{escape(words[0])}</text>\n')
             return
+
         self.out.write(f'<text x="{p.x}" y="{p.y}"{class_str}>\n')
         dy_0 = (len(words) - 1) * (self.cfg.line_spacing * (1 + shift) / 2)
         self.out.write(f'<tspan x="{p.x}" dy="-{dy_0}em">{escape(words[0])}</tspan>')
         for word in words[1:]:
             self.out.write(f'<tspan x="{p.x}" dy="{self.cfg.line_spacing}em">{escape(word)}</tspan>')
         self.out.write("</text>\n")
 
+    def _is_glyph(self, words: Sequence[str]) -> Optional[str]:
+        if len(words) == 1 and (match := self.glyph_name.search(words[0])):
+            return match.group("glyph")
+        return None
+
+    def _draw_glyph(self, p: Point, name: str, cls: Sequence[str], legend_type: str) -> bool:
+        if not self.cfg.glyphs:
+            return False
+
+        if not (glyph := self.cfg.glyphs.get(name)):
+            return False
+
+        if not (view_box := self.view_box_dimensions.match(glyph)):
+            return False
+
+        bound = self.cfg.glyph_tap_size
+        shift = .5
+        if legend_type == "shift":
+            bound = self.cfg.glyph_shift_size
+            shift = 0
+        elif legend_type == "hold":
+            shift = 1
+            bound = self.cfg.glyph_hold_size
+
+        x, y, w, h = tuple([int(v) for v in view_box.groups()])
+
+        # calculate the true dimenions of the image
+        t_w = w - x
+        t_h = h - y
+
+        # confine y to boundary and keep aspect ratio
+        n_h = bound
+        n_w = t_w * (bound/t_h)
+
+        # determine y offset based on position
+        d_y = n_h * shift
+
+        cls = [
```

Comment:
```suggestion
        cls += ["glyph", name]
```
(Couldn't do multiline suggestion)
`cls` is not nullable and should be fine to modify by reference and it is no since we aren't reusing the value after passing it to these methods.

Reply:
It was using destructuring because `cls` is a sequence, which don't necessarily support `__add__`. Probably never the case here but linters aren't happy.

Reply:
Got it, I guess `Sequence` doesn't require that indeed.